instructor/common.py

```diff
@@ -383,23 +402,35 @@ def assignments_edit_treatments(request):
             entries = drug_formset.save(commit=False)
             for obj in drug_formset.deleted_objects:
                 obj.delete()
-            for form in entries:
-                form.assignment = assignment
-                form.save()
+            for drug_instance in entries:
```

Comment:
It looks like we have almost identical code repeated three times. If we had an object that mapped the form prefix to a list of formset and kwarg name to create_treatments, we could DRY it up.

```python
mapping = {'drug': (DrugFormSet, 'drugs'), ...}
for prefix, (ModelForm, treatment_kw) in mapping.items():
  formset = ModelForm(request.POST, prefix=prefix)
  if formset.is_valid():
    entries = formset.save(commit=False)
    for obj in formset.deleted_objects:
      obj.delete()
    for instance in entries:
      if instance.id is None:
        instance.assignment = assignment
        instance.save()
        create_treatments(assignment, **{treatment_kw: instance})
      else:
        instance.save()
```